windows/src/developer/TIKE/child/UfrmPackageEditor.pas

```diff
@@ -1229,15 +1232,25 @@ procedure TfrmPackageEditor.CreateFromCompiledKeyboard(FKMXFilename, FJSFilename
 
   Finally, updates the Keyboards tab
 *)
+
+procedure TfrmPackageEditor.HandlePackageRefreshError(Sender: TObject; msg: string; State: TProjectLogState);
+begin
+  Self.ProjectFile.Project.Log(State, Filename, Msg);
+end;
+
 procedure TfrmPackageEditor.RefreshKeyboardList;
 var
   n, i: Integer;
 begin
+  frmMessages.Clear;
+
   n := lbKeyboards.ItemIndex;
 
   with TPackageInfoRefreshKeyboards.Create(pack) do
   try
-    Execute;
+    OnError := Self.HandlePackageRefreshError;
+    if not Execute then
+      frmMessages.Show;
```

Comment:
I don't see what `frmMessages.Show` displays, but I assume it's an error/feedback message about keyboard version issues?  And that the `Execute` here is `TPackageInfoRefreshKeyboards.Execute`?

Reply:
`frmMessages` is the error message form which should be shown if there are any errors. And yes, `Execute`.